plugins/com.gwtplugins.gwt.eclipse.core/src/com/google/gwt/eclipse/core/dialogs/ModuleSelectionDialog.java

```diff
@@ -1,499 +1,497 @@
+ * Copyright 2011 Google Inc. All Rights Reserved.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package com.google.gwt.eclipse.core.dialogs;
+
+import com.google.gwt.eclipse.core.GWTPlugin;
+import com.google.gwt.eclipse.core.GWTPluginLog;
+import com.google.gwt.eclipse.core.modules.IModule;
+import com.google.gwt.eclipse.core.modules.ModuleFile;
+import com.google.gwt.eclipse.core.modules.ModuleJarResource;
+import com.google.gwt.eclipse.core.modules.ModuleUtils;
+import com.google.gwt.eclipse.core.resources.GWTImages;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.jdt.core.IJarEntryResource;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IPackageFragment;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.internal.corext.util.Strings;
+import org.eclipse.jface.dialogs.IDialogSettings;
+import org.eclipse.jface.viewers.LabelProvider;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.ui.dialogs.FilteredItemsSelectionDialog;
+import org.eclipse.ui.dialogs.SearchPattern;
+import org.eclipse.ui.internal.ide.IDEWorkbenchPlugin;
+import org.eclipse.ui.model.WorkbenchLabelProvider;
+
+import java.text.Collator;
+import java.util.Comparator;
+
+/**
+ * Dialog used for choosing a GWT Module. When the OK button is pressed, an
+ * IModule is returned to the caller.
+ */
+@SuppressWarnings("restriction")
+public class ModuleSelectionDialog extends FilteredItemsSelectionDialog {
+
+  /**
+   * A label provider for details of IModule objects.
+   *
+   * This code was adapted from the
+   * FilteredTypesSelectionDialog.TypeItemDetailsLabelProvider class.
+   */
+  private class ModuleItemDetailsLabelProvider extends ModuleItemLabelProvider {
+
+    @Override
+    public Image getImage(Object element) {
+      if (!(element instanceof IModule)) {
+        return super.getImage(element);
+      }
+
+      Image image = null;
+
+      IModule module = (IModule) element;
+      if (module instanceof ModuleFile) {

```

Comment:
Hmm, while technically true, is this better? I think it might be, but wanted your opinion.

The first thing each block does is to cast it, so that might be right.